### PR TITLE
feat(wallet-mobile): Add dApp Explorer feature flag and rename variables

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves [YOMO-1296](https://emurgo.atlassian.net/browse/YOMO-1296)

[YOMO-1296]: https://emurgo.atlassian.net/browse/YOMO-1296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -151,14 +152,32 @@ export const Dashboard = () => {
 const useNavigateTo = () => {
   const navigation = useNavigation()
 
+  if (CONFIG.DAPP_EXPLORER_ENABLED)
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/Dashboard/Dashboard.tsx]: That's a bit of a pain. To navigate to part of the staking center from outside this dashboard, this condition needs to be replicated?

Comment (michalszorad) [reply apps/wallet-mobile/src/Dashboard/Dashboard.tsx]: Good point, I've refactored this, so the feature flag is not even needed here anymore

**Review by jorbuedo — APPROVED** [review]

File: apps/wallet-mobile/src/features/Discover/useCases/SearchDappInBrowser/SearchDappInBrowserScreen.tsx [hunk 2]
```diff
@@ -9,39 +9,38 @@ import {useBrowser} from '../../common/BrowserProvider'
...
+export const SearchDappInBrowserScreen = () => {
   const {styles} = useStyles()
-  const router = useRoute<RouteProp<BrowserRoutes, 'browser-search'>>()
+  const router = useRoute<RouteProp<BrowserRoutes, 'discover-search-dapp-in-browser'>>()
   const {isEdit} = router.params ?? {isEdit: false}
   const navigateTo = useNavigateTo()
   const {addBrowserTab, setTabActive, updateTab, tabs, tabActiveIndex} = useBrowser()
   const tabActive = tabs[tabActiveIndex]
-  const [searchValue, setSearchValue] = useState('')
+  const [searchValue, setSearchValue] = React.useState('')
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/SearchDappInBrowser/SearchDappInBrowserScreen.tsx:26]: heads up: we have a hook + provider that transforms the navbar into a searchable bar. (dunno if it's applied on nav bar this one)

File: apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx [hunk 3]
```diff
@@ -16,11 +16,11 @@ export const BrowserToolbar = ({uri}: Props) => {
   const {isSecure, domainName} = getDomainFromUrl(uri)
 
   const handleCloseBrowser = () => {
-    navigateTo.discover()
+    navigateTo.selectDappFromList()
   }
 
   const handleEditUrl = () => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx:22]: handle edit and navigates to search, is that correct? Guessing: if the user taps on location we navigate to a search?

File: apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx [hunk 4]
```diff
@@ -16,11 +16,11 @@ export const BrowserToolbar = ({uri}: Props) => {
   const {isSecure, domainName} = getDomainFromUrl(uri)
 
   const handleCloseBrowser = () => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx:18]: same here, if the browser is closed so the user is moved back to select screen?

File: apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx (hunk 3 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx:22]: Correct

File: apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx (hunk 4 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserToolbar.tsx:18]: Correct

File: apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserTabsBar.tsx [hunk 5]
```diff
@@ -23,7 +23,7 @@ export const BrowserTabsBar = () => {
 
   const handleCreateTab = () => {
     switchTab(false)
-    navigateTo.browserSearch(false)
+    navigateTo.searchDappInBrowser({isEdit: false})
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserTabsBar.tsx]: params are unsafe, later if possible move to state

File: apps/wallet-mobile/src/features/Discover/common/useNavigateTo.tsx [hunk 6]
```diff
@@ -1,24 +1,18 @@
...
+      return navigation.navigate('discover-browser', {screen: 'discover-search-dapp-in-browser', params: {isEdit}})
     },
-    discover: () => navigation.navigate('discover-list'),
-    browserView: () =>
-      navigation.navigate('browser', {
-        screen: 'browser-view',
-      }),
+    selectDappFromList: () => navigation.navigate('discover-select-dapp-from-list'),
+    browseDapp: () => navigation.navigate('discover-browser', {screen: 'discover-browse-dapp'}),
     navigation: () => navigation,
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/common/useNavigateTo.tsx]: this should be deleted.

Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/common/useNavigateTo.tsx]:
```suggestion
```

File: apps/wallet-mobile/src/features/Discover/common/BrowserProvider.tsx [hunk 7]
```diff
@@ -106,7 +106,7 @@ export const BrowserProvider = ({
 }
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/common/BrowserProvider.tsx:97]: @michaeljscript this is looking confusing and with a lot of var shadowing, can you give it a look?

File: apps/wallet-mobile/src/features/Discover/useCases/SearchDappInBrowser/SearchDappInBrowserScreen.tsx (hunk 2 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Discover/useCases/SearchDappInBrowser/SearchDappInBrowserScreen.tsx:26]: We have the hook used in `SelectDappFromListScreen`, but this component is the search in the browser, which has a different UI, so I'd say it's okay to use a custom component

File: apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserTabsBar.tsx (hunk 5 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Discover/useCases/BrowseDapp/BrowserTabsBar.tsx]: I'll refactor this. It looks like we don't need params nor state

File: apps/wallet-mobile/src/features/Discover/common/BrowserProvider.tsx (hunk 7 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Discover/common/BrowserProvider.tsx:97]: Updated

**Pull request merged**